domain/secret/state/state.go

```diff
@@ -191,14 +347,52 @@ ON CONFLICT(id) DO UPDATE SET
 	return nil
 }
 
-func (st State) upsertSecretModelOwner(ctx context.Context, tx *sqlair.TX, owner secretOwner) error {
+func (st State) upsertSecretModelOwner(ctx context.Context, tx *sqlair.TX, owner secretModelOwner) error {
 	insertQuery := `
 INSERT INTO secret_model_owner (secret_id, label)
-VALUES      ($secretOwner.*)
+VALUES      ($secretModelOwner.*)
 ON CONFLICT(secret_id) DO UPDATE SET label=excluded.label
 `
 
-	insertStmt, err := st.Prepare(insertQuery, secretOwner{})
+	insertStmt, err := st.Prepare(insertQuery, secretModelOwner{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = tx.Query(ctx, insertStmt, owner).Run()
```

Comment:
Handle unique constraint error then return label already exist error.

Reply:
No need - this is already done via checkUserSecretLabelExists in the same txn